stronk/schemas/user/query.py

```diff
@@ -1,36 +1,42 @@
 import graphene
+from flask import g
 
+from stronk.constants import USER_NOT_FOUND_MSG
+from stronk.errors.not_found import NotFound
 from stronk.models.user import User as UserModel
+from stronk.schemas.protected_user.type import ProtectedUser
 from stronk.schemas.user.type import User
+from stronk.schemas.user_interface.type import UserInterface
 
 
 class Query(graphene.ObjectType):
-    users = graphene.List(User)
-    user = graphene.Field(lambda: User,
-                          id=graphene.String(),
-                          username=graphene.String(),
-                          email=graphene.String(),
-                          currentProgram=graphene.Int())
+    user = graphene.Field(
+        UserInterface, username=graphene.String(required=True))
+    users = graphene.List(UserInterface)
 
     def resolve_users(root, info):
-        query = User.get_query(info)
-        return query.all()
-
-    def resolve_user(root, info, id=None, username=None, email=None, currentProgram=None):
-        """Search for user in decreasing order of precedence: id, username,
-        email and current program.
+        """Returns UserInterface[] information about all users.
+        Other users are type ProtectedUser whereas the user itself is type User.
         """
-        query = User.get_query(info)
-        if id:
-            return query.filter(UserModel.id == id).first()
-
-        if username:
-            return query.filter(UserModel.username == username).first()
-
-        if email:
-            return query.filter(UserModel.email == email).first()
-
-        if currentProgram:
-            return query.filter(UserModel.current_program == currentProgram).first()
+        users = User.get_query(info).all()
+        result = []
+        for user in users:
+            if user.id == g.id:
+                result.append(user)
+            else:
+                result.append(ProtectedUser(
```

Comment:
Ah mb I got it the wrong way i meant - I meant method on protected user to convert users, but yup converter sounds good too. Do we need to convert from `ProtectedUser` back to `User`?

Reply:
I'm thinking that we actually probably don't need to be implementing a converter from `ProtectedUser` to `User` because this functionality isn't currently required right?